tools/cli_commands/cost_report/cost_management_api.py

```diff
@@ -1,40 +1,44 @@
-from typing import Any, List, Self
+from typing import List
+
+from requests.adapters import HTTPAdapter
+from urllib3.util import Retry
 
 from reconcile.utils.oauth2_backend_application_session import (
     OAuth2BackendApplicationSession,
 )
+from reconcile.utils.rest_api_base import ApiBase
 from tools.cli_commands.cost_report.response import ReportCostResponse
 
 REQUEST_TIMEOUT = 60
 
 
-class CostManagementApi:
+class CostManagementApi(ApiBase):
     def __init__(
         self,
         base_url: str,
         token_url: str,
         client_id: str,
         client_secret: str,
         scope: List[str] | None = None,
-        request_timeout: int | None = REQUEST_TIMEOUT,
     ) -> None:
-        self.base_url = base_url
-        self.request_timeout = request_timeout
-        self.session = OAuth2BackendApplicationSession(
+        session = OAuth2BackendApplicationSession(
             client_id=client_id,
             client_secret=client_secret,
             token_url=token_url,
             scope=scope,
         )
-
-    def __enter__(self) -> Self:
-        return self
-
-    def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None:
-        self.cleanup()
-
-    def cleanup(self) -> None:
-        self.session.close()
+        for prefix in ["http://", "https://"]:
```

Comment:
Do you mind and bring this into `ApiBase.__init__`?

Reply:
sure, that requires some changes in `ApiBase` as currently it will be ignored once `session` passed in.

Reply:
updated at https://github.com/app-sre/qontract-reconcile/pull/4268/commits/034419680c2e6d695033008eb4a62e02386a70f2, please have a look

Reply:
Great! 🚢 it